Skeleton of profile, favorites, ratings

midorili opened the pull request.

midorili committed:
Skeleton of profile, favorites, ratings

midorili committed:
Slight tweak to getUserInfo

kahlstorf1 commented on app/db.ts:
put supabase URL in .env

kahlstorf1 commented on app/user/[userID]/classes/page.tsx:
Fix Linting Errors and comment out UseState until we use it (will cause linting errors if it is imported but not used)

kahlstorf1 commented on app/user/[userID]/favorites/page.tsx:
```diff
@@ -0,0 +1,32 @@
+        <div className="inline-block">
+          Placeholder 1
+        </div>
+        <br></br>
```
linting fix for empty components - change `<br></br> to <br />`

kahlstorf1 replied:
fix linting errors (indentation)

kahlstorf1 commented on app/user/[userID]/page.tsx:
Linting Errors from importing components not used, import Next/link before importing indiv. components

danny-avila replied:
resolved in my commit

kahlstorf1 commented on app/user/[userID]/profile/page.tsx:
```diff
@@ -0,0 +1,95 @@
+
+  const [user, setUser] = useState<Array<User> | null>([]);
+  // const [user, setUser] = useState<User | null>(null);
+  async function getUserInfo() {
```
Type Error because function is not returning anything, can use linter ignore next line

kahlstorf1 commented on app/user/[userID]/ratings/page.tsx:
`label htmlFor`
`input id`

to get rid of linting (A form label must be associated with a control) errors

kahlstorf1 commented on app/user/[userID]/ratings/page.tsx:
```diff
@@ -0,0 +1,71 @@
+            <label>4</label>
+          <input name="rating" type="radio" value="5"></input>
+            <label>5</label>
+          <button className="ml-5 rounded-lg border-black border-2">Submit</button>
```
add type to button

kahlstorf1 commented on app/user/[userID]/ratings/page.tsx:
```diff
@@ -0,0 +1,71 @@
+            <label>4</label>
+          <input name="rating" type="radio" value="5"></input>
+            <label>5</label>
+          <button className="ml-5 rounded-lg border-black border-2">Submit</button>
```
add type to button

kahlstorf1 commented on components/ClassesMock.tsx:
disable linting error for favorite not being used until we use

kahlstorf1 commented on components/Favorites.tsx:
same empty component error fixes as /favorites

kahlstorf1 commented on components/Ratings.tsx:
same empty component error fixes as /favorites

danny-avila committed:
refactor(page.tsx): remove async from UserProfile function definition
feat(page.tsx): add defaultUser object to initialize user state
feat(page.tsx): change user state from array to object
feat(page.tsx): update setUser to set user object instead of array
feat(page.tsx): add useEffect to log user state changes

midorili committed:
Some linter fixes

midorili committed:
Merge branch 'MLUserProfile' of https://github.com/rpp2210-BOC-Aquaforce/FitnessPass into MLUserProfile

midorili committed:
Fixed all linter issues

midorili committed:
Added some css

danny-avila committed:
Merge branch 'main' into MLUserProfile
